lib/absinthe/relay/connection.ex

```diff
@@ -3,10 +3,10 @@ defmodule Absinthe.Relay.Connection.Options do
 
   @typedoc false
   @type t :: %{
-          optional(:after) => nil | integer,
-          optional(:before) => nil | integer,
-          optional(:first) => nil | integer,
-          optional(:last) => nil | integer
+          optional(:after) => Connection.cursor(),
+          optional(:before) => Connection.cursor(),
+          optional(:first) => pos_integer(),
+          optional(:last) => pos_integer()
```

Comment:
Notably if we're going to make these `nil` then I think we need to set defaults on the struct, otherwise dialyzer will be unhappy.

Reply:
Not sure if I understand. Do You want the types to be `nil | Something`? 
Isn't that covered by the `optional` type? I'm new to these typespecs, so I'm not sure if I'm reading them correctly?

Reply:
Ah, is the optional only to mark that the key might not be there? 
So for a type signature of  `optional(:foo) => nil | integer`, that would mean that the key might not be there, and if it is, it's either `nil` or an `integer`? 

I'll defer to Your decision, I'd just like it for the `after` and `before` to stop being `integer`s

Reply:
> Ah, is the optional only to mark that the key might not be there?
That is correct, the `optional(:foo) => nil | integer` says that `:foo` may or may not be present, and it's value may be `nil` or an `integer` if it is there.

Let's stick with `nil |` for now.